common4j/src/main/com/microsoft/identity/common/java/providers/nativeauth/responses/signup/SignUpChallengeApiResponse.kt

```diff
@@ -0,0 +1,168 @@
+    @SerializedName("challenge_type") val challengeType: String?,
+    @Expose @SerializedName("binding_method") val bindingMethod: String?,
+    @Expose @SerializedName("interval") val interval: Int?,
+    @SerializedName("challenge_target_label") val challengeTargetLabel: String?,
+    @Expose @SerializedName("challenge_channel") val challengeChannel: String?,
+    @SerializedName("signup_token") val signupToken: String?,
+    @Expose @SerializedName("error") val error: String?,
+    @Expose @SerializedName("error_description") val errorDescription: String?,
+    @SerializedName("details") val details: List<Map<String, String>>?,
+    @Expose @SerializedName("code_length") val codeLength: Int?
+) : IApiResponse(statusCode) {
+
+    companion object {
+        private val TAG = SignUpChallengeApiResponse::class.java.simpleName
+    }
+
+    /**
+     * Maps potential errors returned from the server response, and provide different states based on the response.
+     * @see com.microsoft.identity.common.java.providers.nativeauth.responses.signup.SignUpChallengeApiResult
+     */
+    fun toResult(): SignUpChallengeApiResult {
+        LogSession.logMethodCall(TAG, "${TAG}.toResult")
+
+        return when (statusCode) {
+
+            // Handle 400 errors
+            HttpURLConnection.HTTP_BAD_REQUEST -> {
+                return when {
+                    error.isUnsupportedChallengeType() -> {
+                        SignUpChallengeApiResult.UnsupportedChallengeType(
+                            error = error.orEmpty(),
+                            errorDescription = errorDescription.orEmpty(),
+                            details = details,
+                        )
+                    }
+                    error.isExpiredToken() -> {
+                        SignUpChallengeApiResult.ExpiredToken(
+                            error = error.orEmpty(),
+                            errorDescription = errorDescription.orEmpty(),
+                            details = details,
+                        )
+                    }
+                    else -> {
+                        SignUpChallengeApiResult.UnknownError(
+                            error = error.orEmpty(),
+                            errorDescription = errorDescription.orEmpty(),
+                            details = details,
+                        )
+                    }
+                }
+            }
+
+            // Handle challenge types
+            HttpURLConnection.HTTP_OK -> {
+                return when {
+                    challengeType.isRedirect() -> {
+                        SignUpChallengeApiResult.Redirect
+                    }
+                    challengeType.isOOB() -> {
+                        return when {
+                            challengeTargetLabel.isNullOrBlank() -> {
+                                SignUpChallengeApiResult.UnknownError(
+                                    error = "invalid_state",
```

Comment:
I'm sure these errors are defined as constants in our code. Please leverage those constants; here and everywhere

Reply:
I have created a constant in ApiErrorResult.kt file. There was no pre-existing constant for this value.